src/models/karaoke.py

```diff
@@ -0,0 +1,220 @@
+from dataclasses import dataclass
+from datetime import timedelta
+from functools import reduce
+from itertools import accumulate
+from typing import Sequence
+
+from src.models import SongLine
+
+from src.fonts import FontScaler
+
+import pyass
+from pyass import Style
+
+
+class KLine:
+    pass
+
+
+class KSyl:
+    pass
+
+
+@dataclass
+class KChar:
+    char: str
+    i: int
+    sylI: int
```

Comment:
What are these variables?

Reply:
`i`: character's index in the line
`sylI`: character's index in the syllable